File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/DiagnosisKeySubmissionToken.java
package uk.nhs.nhsx.virology;

import uk.nhs.nhsx.core.ValueType;

import static uk.nhs.nhsx.core.Preconditions.checkArgument;

public class DiagnosisKeySubmissionToken extends ValueType<DiagnosisKeySubmissionToken> {

    private DiagnosisKeySubmissionToken(String value) {
        super(value);
        checkArgument(!value.isEmpty(), "Diagnosis key submission token not present");
    }

    public static DiagnosisKeySubmissionToken of(String value) {
        return new DiagnosisKeySubmissionToken(value);
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/DiagnosisKeySubmissionToken.kt
package uk.nhs.nhsx.virology

import dev.forkhandles.values.NonBlankStringValueFactory
import dev.forkhandles.values.StringValue

class DiagnosisKeySubmissionToken private constructor(value: String) : StringValue(value) {
    companion object : NonBlankStringValueFactory<DiagnosisKeySubmissionToken>(::DiagnosisKeySubmissionToken)
}
